CRM/Utils/Check.php

```diff
@@ -209,7 +212,7 @@ public static function checkAll($max = FALSE) {
 
     foreach ($messages as $key => $message) {
       $hush = self::checkHushSnooze($message);
-      $message->setVisible(!$hush);
+      $messages[$key]->setVisible(!$hush);
```

Comment:
No, deleting it broke things.  I made the commit here, saw that everything worked, got thinking about how you pointed out the API was working fine, thought `$message->setVisible(!$hush);` wouldn't do anything, deleted it, and everything broke.  I then abandoned the further changes.  Basically when I said "somehow that this is the actual $messages[$key] instance rather than a copy of $message that'll get left behind as you're foreaching", I was discovering the phenomenon that you and @totten explained but not knowing exactly what I was describing.